### PR TITLE
support synonym functions

### PR DESCRIPTION
**Pull request by luotao24**

### CODE REVIEW

**Comment by luotao24** [comment]
#57

File: contrib/postgres_fdw/postgres_fdw.c [hunk 1]
```diff
@@ -1587,6 +1597,63 @@ postgresBeginForeignScan(ForeignScanState *node, int eflags)
...
+	 * Get connection to the foreign server.  Connection manager will
+	 * establish new connection if necessary.
+	 */
+	fsconn = GetConnection(user, false, &fsstate->conn_state);
+
+	/* Assign a unique ID for my cursor */
+	fsstate->cursor_number = GetCursorNumber(fsstate->conn);
+	fsstate->cursor_exists = false;
+
+	query = palloc(strlen("SELECT * FROM .") + strlen(schemaname) + strlen(relname) + 10);
```
Comment (BingAo-Office) [inline contrib/postgres_fdw/postgres_fdw.c:1627]: did you considered multi-byte character ?

File: contrib/postgres_fdw/postgres_fdw.c [hunk 2]
```diff
@@ -1587,6 +1597,63 @@ postgresBeginForeignScan(ForeignScanState *node, int eflags)
...
+	sprintf(query, "SELECT * FROM %s.%s", schemaname, relname);
+	fsstate->query = query;
+
+	last_result = PQexec(fsconn, fsstate->query);
+
+	*numattr = PQnfields(last_result);
+
+	pfree(fsstate);
```
Comment (BingAo-Office) [inline contrib/postgres_fdw/postgres_fdw.c:1635]: fsstate->query  need free also.

File: contrib/postgres_fdw/postgres_fdw.c [hunk 3]
```diff
@@ -1587,6 +1597,63 @@ postgresBeginForeignScan(ForeignScanState *node, int eflags)
...
+	ReleaseConnection(fsconn);
+
+	return (void *)last_result;
```
Comment (BingAo-Office) [inline contrib/postgres_fdw/postgres_fdw.c:1638]: first, PQresult need  to clear by PQclear（）。 but， I  can't find PQclear anywhere.
second,  PQconnection need to close by PQfinish() also.

**Review by BingAo-Office — CHANGES_REQUESTED** [review]

**Pull request closed without merging**